internal/worker/bootstrap/worker.go

```diff
@@ -206,6 +208,38 @@ func (w *bootstrapWorker) loop() error {
 	if !ok {
 		return errors.Errorf("state serving information not available")
 	}
+
+	// Fetch spaces from substrate.
+	// We need to do this before setting the API host-ports,
+	// because any space names in the bootstrap machine addresses must be
+	// reconcilable with space IDs at that point.
+	callContext := envcontext.WithoutCredentialInvalidator(ctx)
+	env, err := getEnviron(ctx, w.cfg.SystemState, w.cfg.CloudService, w.cfg.CredentialService, w.cfg.NewEnviron)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// TODO(nvinuesa): Fans should be retrieved from the model config
+	// service once it's finished instead of legacy state.
+	model, err := w.cfg.SystemState.Model()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	modelConfig, err := model.Config()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	fanConfig, err := modelConfig.FanConfig()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if err = space.ReloadSpaces(callContext, w.cfg.SystemState, w.cfg.SpaceService, env, fanConfig); err != nil {
+		if !errors.Is(err, errors.NotSupported) {
+			return errors.Trace(err)
+		}
+		w.cfg.LoggerFactory.Child("").Debugf("Not performing spaces load on a non-networking environment")
```

Comment:
This seems invalid. Just get a logger from the factory and put it in the constructor.